src/client/opamPinCommand.ml

```diff
@@ -366,6 +366,43 @@ let default_version st name =
   try OpamPackage.version (OpamSwitchState.get_package st name)
   with Not_found -> OpamPackage.Version.of_string "~dev"
 
+let fetch_all_pins st pins =
+  let root = st.switch_global.root in
+  let fetched =
+    let cache_dir =
+      OpamRepositoryPath.download_cache OpamStateConfig.(!r.root_dir)
+    in
+    let command (name, url, subpath) =
+      let srcdir =
+        OpamPath.Switch.pinned_package root st.switch
+          (OpamPackage.Name.of_string name)
+      in
+      OpamProcess.Job.Op.(
+        OpamRepository.pull_tree ~cache_dir ?subpath name srcdir [] [url]
+        @@| fun r -> (name, url, subpath, r))
+    in
+    OpamParallel.map ~jobs:OpamStateConfig.(!r.dl_jobs) ~command pins
+  in
+  let errored, to_pin =
+    List.fold_left (fun (err,ok) result ->
+        let name, url, subpath, result = result in
+        match result with
+        | Not_available _ -> (* clean dir ? *)
+          (name, url, subpath)::err, ok
+        | _ -> err, (url, subpath)::ok)
+      ([],[]) fetched
+  in
+  if errored = []
+  || OpamConsole.confirm "Could not retrieve%s\nContinue?"
```

Comment:
We may want to be more clear about the implications of saying "yes" here ?
Also, be careful that `yes` is the default and may be forced with `--yes` misled people.

Reply:
Maybe adding something like "You may reach an inconsistent state because of those not pinned packages"?

Reply:
updated